cmd/worker/worker.go

```diff
@@ -201,10 +203,10 @@ func (c *Command) Start(ctx context.Context) error {
 				Workloads:     true,
 				SingleNode:    false,
 				K0sVars:       c.K0sVars,
-				ClusterConfig: clusterConfig,
+				ClusterConfig: nil,
```

Comment:
I'm not sure if it's necessary to have the config in `k0s status -o json` on a worker. I think it used to work on main, here it is just nil. I set the nil explicitly just to be clear about it :)

Reply:
The problem with having it there is that then the status component would need to be reconcilied, but there's no reconciliation on a worker or in the nodecomponents-manager on a controller where the status component is hosted, so this would be (has been) showing just the first dynamic config unless a bigger change is made. 

The function in main has been doing this:

```go
		clusterConfig, err := config.LoadClusterConfig(c.K0sVars)
		if err != nil {
			return fmt.Errorf("failed to load cluster config: %w", err)
		}
```
==>
```go
func LoadClusterConfig(k0sVars constant.CfgVars) (*v1beta1.ClusterConfig, error) {
	loadingRules := ClientConfigLoadingRules{K0sVars: k0sVars}
	return loadingRules.Load()
}
```
==>
```go
func (rules *ClientConfigLoadingRules) Load() (*v1beta1.ClusterConfig, error) {
	if rules.Nodeconfig {
		return rules.fetchNodeConfig()
	}
	if !rules.IsAPIConfig() {
		return rules.readRuntimeConfig()
	}
	if rules.IsAPIConfig() {
		nodeConfig, err := rules.BootstrapConfig()
		if err != nil {
			return nil, err
		}
		apiConfig, err := rules.ClusterConfig()
		if err != nil {
			return nil, err
		}
		// get node config from the config-file and cluster-wide settings from the API and return a combined result
		return rules.mergeNodeAndClusterconfig(nodeConfig, apiConfig)
	}
	return nil, nil
}
```


So, it seems the config has been nil on a worker all along.

For controller it works like it used to, it shows the config k0s was started with, never the dynamic config.

IMO changing this behavior can be done in a separate PR as it works here as it has worked all along.